packages/ses/README.md

```diff
@@ -503,6 +502,153 @@ system generated other diagnostic information hidden in side tables. The tamed
+
+However, such a program can:
+
+- execute for an indefinite amount of time,
+- allocate arbitrary amounts of memory,
+- detect the platform endianness,
+- in some JavaScript engines, observe the contents of the stack.
+  This may include sensitive information about the layout of files on the host
+  disk.
+  In cases where the stack is data-dependent, a guest can infer the data.
+  SES occludes the stack on V8 and SpiderMonkey, but can not on
+  JavaScriptCore.
+
+```js
+// Claims, Figure 1
+lockdown();
+const compartment = new Compartment();
+compartment.evaluate(program);
+```
+
+The trusted compute base includes:
+
+- the host hardware,
+- the host operating system,
+- the process memory manager,
+- a minimal implementation of JavaScript conforming to ECMAScript 262 as of
+  2021, providing no unspecified host behavior like the introduction of syntax
+  that when evaluated reveals a mutable object.
+  SES accounts for one such host behavior provided by Node.js, namely the `domain`
+  property on promises, by preventing the use of `ses` in concert with the
+  `domain` module.
+- Also, any attached debugger, and
+- any JavaScript that has executed in the same realm before the host calls
+  `lockdown`, including JavaScript that executes after `ses` initializes.
+
+If the host program (Figure 2) arranges for the compartment's `globalThis` to
+be frozen, we additionally claim that the host can evaluate any two programs
+(`program1` and `program2`) such that neither program will:
+
+- initially share *any* mutable objects.
+- be able to observe the relative passage of time of the other program,
+- specifically because they do not have access to any shared mutable state as
+  they would had they been given a reference to `Math.random()`.
+
+```js
+// Claims, Figure 2
+lockdown();
+const compartment = new Compartment();
+harden(compartment.globaThis);
+compartment.evaluate(program1);
+compartment.evaluate(program2);
+```
+
+However no such programs (`program`, `program1`, nor `program2`) can be useful
+as such.
+A host program is therefore responsible for maintaining any of the desired
+invariants above when "endowing" a compartment with any of its own objects.
+
+For example, a host program (Figure 3) may run two programs in separate
+compartments, giving one program the ability to resolve a promise and the other
+program the ability to observe the resolution.
+The host program is responsible for hardening these objects.
+
+With such an arrangement, the host and both guest programs are vulnerable
+to denial of service.
+
+```js
+// Claims, Figure 2
+lockdown();
+
+const promise = new Promise(resolve => {
+  const compartmentA = new Compartment(harden({
+    resolve,
+  }));
+  compartment.evaluate(programA);
+});
+
+const compartmentB = new Compartment(harden({
+  promise,
+}));
+compartmentB.evaluate(programB);
+```
+
+Any agent can also hide their own stack by causing client callbacks to
```

Comment:
Why are we switching the terminology to "agent" ?

Reply:
Hard to find a word that isn’t overloaded, neither agent is necessarily the host, both might be guests connected by introduction. Maybe I can work a clarification in.

Reply:
subsystem

Reply:
I seem to have found a way to use “any other guest or the host” instead of “agent”.